### PR TITLE
Change Segment integration to respect DoNotTrack

### PR DESCRIPTION
**Pull request by atduarte**
Similar condition as the one we have for Google Analyics.

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/756"></a>

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/756"></a>

### CODE REVIEW

File: src/components/segment.svelte [hunk 1]
```diff
@@ -88,9 +95,15 @@
     analytics._writeKey = writeKey;
     // Add a version to keep track of what's in the wild.
     analytics.SNIPPET_VERSION = "4.13.2";
-    // Load Analytics.js with your key, which will automatically
-    // load the tools you've enabled for your account. Boosh!
-    analytics.load(writeKey);
+
+    if (!isDoNotTrack()) {
```
Comment (mikenikles) [inline src/components/segment.svelte:99]: Nitpick: I think it'd be more readable to remove `!` and switch the if/else blocks.

**Review by mikenikles — APPROVED** [review]
A small nitpick on readability. Good to go from my end.

File: src/components/segment.svelte [hunk 2]
```diff
@@ -16,6 +16,13 @@
       ? "5aJzy2ASNbqx8I0kwppRflDZpL7pS1GO" // Website Production
       : "Xe5zR3MbnyxHsveZr4HvrY35PL9iT0EH"; // Website Staging
 
+  const isDoNotTrack = () =>
+    typeof navigator !== "undefined" &&
+    (parseInt(navigator.doNotTrack) === 1 ||
+      parseInt(window.doNotTrack) === 1 ||
+      parseInt(navigator.msDoNotTrack) === 1 ||
+      navigator.doNotTrack === "yes");
```
Comment (jakobhero) [inline src/components/segment.svelte:24]: does it ever hold this value?

**Review by jakobhero — APPROVED** [review]
LGTM

**Comment by mikenikles** [comment]
@atduarte What's the status of this PR?

File: src/components/segment.svelte (hunk 2 above)
Comment (atduarte) [reply src/components/segment.svelte:24]: I'm not sure, as I used what existed already for Google Analytics.

**Comment by atduarte** [comment]
> @atduarte What's the status of this PR?

I thought I had merged this already. Doing it now.

**Pull request merged**